identity-service/src/userHandleMiddleware.js

```diff
@@ -0,0 +1,64 @@
+const axios = require('axios')
+const { sendResponse, errorResponseBadRequest } = require('./apiHelpers')
+
+const models = require('./models')
+
+const audiusLibsWrapper = require('./audiusLibsInstance')
+
+/**
+ * queryDiscprovForWalletAddressAndUserId - Queries the discovery provider for the user w/ the handle
+ * @param {string} handle
+ * @returns {object} User Metadata object
+ */
+const queryDiscprovForWalletAddressAndUserId = async (handle) => {
+  const { discoveryProvider } = audiusLibsWrapper.getAudiusLibs()
+
+  const response = await axios({
+    method: 'get',
+    url: `${discoveryProvider.discoveryProviderEndpoint}/users`,
+    params: { handle }
+  })
+
+  const usersList = response.data.data
+  if (!Array.isArray(usersList) || !(usersList.length >= 1)) {
+    throw new Error('Unable to retrieve user from discovery provider')
+  }
+
+  return usersList[0]
+}
+
+/**
+ * user handle middleware:
+ * make sure a handle was passed as a query parameter
+ * if user handle is in database, proceed
+ * otherwise, query for the user with that handle in DP
+ * if DP does not return a user or if wallet address of returned user is not found in the database, bad request
+ * otherwise, update user for that wallet address with the handle and DP user id, then proceed
+ */
+async function userHandleMiddleware (req, res, next) {
+  try {
```

Comment:
I dont know if it "needs" but it would be good to also include the handle from DP if it's absent. authmiddleware logic also has bits of logic related to recovering signatures. can make the change to handle missing user handles but maybe as a separate PR? doesn't seem super related here